### PR TITLE
Nukies Update 7: Hats (Also massive uplink standardization, weapon kits and ammo changes)

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Massively overhauls and standardizes the nuclear operative uplink. 

### Weapon Kits

Essentially, all the main weapons of the uplink have been changed to instead come as 'weapon kits', which are essentially cases containing a weapon loadout to enable operatives to easily start operating on only just one item purchase, without the fuss of worrying whether or not operatives are getting spare ammo, or getting relevant equipment for success. Consider this a pseudo-loadout, though without necessarily restricting the purchasing of more weapon kits. 

All kits come in three categories: Low Cost (8 TC), Medium Cost (14 TC) and High Cost (18 TC). This is also matched by categorized ammo costs; Basic Ammo (2 TC), Hollow Point and Armour Penetrating (4 TC), Incendiary (3 TC) and Special (or anything that does not easily fit these categories and does something real extra) (5 TC). Weapons that lacked these ammos have gained these ammo types to fill the gaps. 

<details>
There is may one exception to this in disruptor ammo, which is priced as basic ammo if only because it isn't _quite_ good enough to justify pricing at 5 tc and I can see an op wanting to use it as a basic ammo type instead of normal .50 BMG against, say, a silicon/mech heavy opposition. Since it cannot kill organics on its own, I'll consider this mostly basic-adjacent
</details>
The kits have also been labelled based on potential difficulty. This reflects possible difficulties in using the item, how conducive it is to success for how much game knowledge needed to actually use it, and how likely an op is to succeed using it. I don't expect ops to win using nothing but a rocket launcher, but I think ops should get a fair shake at trying, yeah?

The kits are as below:
#### **Low-Cost**
_Bulldog (Moderate):_ Shotgun and three magazines of standard ammo.
_Ansem (Easy/Spare):_ Pistol and three spare magazines of standard ammo.
#### **Medium Cost**
_C-20r (Easy):_ SMG and three spare magazines of standard ammo.
_Energy Sword and Shield (Very Hard):_ Energy sword and shield. (Also a special hat)
_Revolver (Moderate):_ Revolver and three speedloaders of standard ammo.
_Rocket Launcher (Hard):_ Rocket launcher with three spare rockets.
#### **High Cost**
_L6 SAW (Moderate):_ LMG, and that's it. No spare ammo.
_M-90gl (Hard):_ Rifle, two spare magazines of standard ammo and a box of rubber grenades.
_Sniper (Hard):_ Sniper rifle, two spare magazine of standard ammo, and one magazine of disruptor ammo. Also suit and tie.
_CQC (Very Hard):_ Comes with a stealth implant and a bandana.
_Double-Energy Sword (Very Hard):_ Double-energy sword, syndicate soap, antislip module, meth injector and a prisoner jumpsuit.
_**NEW** Grenadier's Kit (Hard):_ Grenadier's belt and grenade launcher (the one that launchers chem grenades). (I replaced the shit acid grenade with another flashbang in the belt)

Surplus SMG (Flukie difficulty) has been unchanged. It just now comes with two rations.

Includes two new revolver ammo types: Phasic, which goes through walls and armor, but has significantly less damage as a result (I've equalized the revolver damage and the rifle version's damage to 30 for both). And Heartseeker, which has homing bullets. Both are Special ammo, and are priced at 5 TC a speedloader.

### Other Gear

The other items in the uplink have also been consolidated and standardized in various ways.

#### Grenades

Most now cost 15 TC for three grenades of any given type (including the full fungal tuberculous). This is pretty much identical to the previous price, just more consistent overall and front-loaded in cost.

#### Reinforcements

All the various reinforcements now cost 35 TC and all refundable, equalizing cost to the average across the reinforcements. This is primarily because I feel like all these options should be weighed equally, and not one of these options are necessarily worse or better than the other in their current balance. They're largely inaccessible for normal ops regardless, and typically come out when there is a discount or war ops. I took the average value and went with it. Not much more to say.

#### Mechs

They're just cheaper. These things still suck and they need help. They've always needed help. A slightly less excessive value for the mechs may help see people willing to spend the TC on them. I doubt it. I seriously suggest not buying these still. I keep them in primarily because they are big stompy mechs and are kind of iconic 'war ops' gear.

#### Bundles

Since I've implemented weapon kits, gun bundles are rather redundant. So the bulldog weapon and ammo bundle, the c20-r weapon and ammo bundle and technically the sniper bundle were removed. The sniper bundle is now the weapon kit, obviously.

Nothing else here really. Except for one....

#### Implants

Not much changed here. I standardized the implant prices to 8 TC a pop. This is in accordance with traitor implants, which ops also get. So everything in this category bar a few exceptions (like macro/microbombs) are around 8 TC. Makes sense to me, really.

Importantly, I made the Implant bundle 25 TC, and I unrandomized the contents. Who in the right fucking mind would spend 40 TC just to get five reviver implants is beyond me. But instead, you get one of each of the cybernetic implants except thermal eyes (you can just buy thermals and get the benefit of both vision types; x-ray and thermal vision, if you want to use smokescreens a lot).

#### Base Keys

They're all now 15 TC, except the fridge which is 5 TC. It's weird they're valued differently when they are taken mostly to do gimmicks like xenobio and toxins in a hurry before hitting the station. So we've standardized it.

## Hat Crate

**YES, GOOD SIR, YOU TOO CAN ORDER A HAT CRATE FROM THE SYNDICATE STORE FOR ONLY 5 TC!**

**NO NEED FOR A KEY, JUST BUY IT AND PULL IT OPEN WITH YOUR STANDARD ISSUE CROWBAR!**

**ENJOY YOUR NEW CRATE! ENJOY YOUR NEW HAT!**

**PUT IT ON USING THE FREE HAT STABILIZERS WE INCLUDED WITH THE HATS!**

~~**NO REFUNDS IF YOU GET BLOOD ON YOUR HAT!**~~

<details>
There is a 1% chance to instagib people with direct hits from a rocket. This does the crit effect.
</details>

## Why It's Good For The Game

The uplink needed more spring cleaning and standardization.

With this, I've partially implemented my older idea for ammo consistency and initial allowance for nukies. Ammo is kind of over-priced and often where a good chunk of TC goes towards without really pushing nukies towards meaningful success. And it is often what is tripping up new players who didn't think to get any. Now, when they get a gun, they get ammo in their case. On top of this, the weapon kit category is both at the top of the uplink AND has a little label to say 'Recommend', so that these new players will hopefully know they should be looking there first.

In addition, it is the gateway towards a concept that is currently being worked on. Nuclear operatives having some degree of predefined loadouts for players to select if they aren't sure what they want, or don't know what to get. Nukies is very confusing for many players. So giving them a fighting chance with some premade setups can help ease them into the role without needing too much player knowledge in how to apply the items. This is only one step towards that, so that players can identify what gear they need to help succeed based on their skill.

I wanted to implement a difficulty warning so that players can choose gear loadouts that are actually conducive to their skill and knowledge. I based it on how much players would need to know to engage in combat with it, and how much fiddling is required to get something to work properly (overly involved reloading is a consideration, for example, as well as precise button presses). In addition, how much of a force multiplier some weapons can be for their ease of use.

Most people recognize the c20-r as the most new player friendly weapon, as an example. So it would be good to steer players towards taking that gun because of how easy it is to use, understand and succeed with it.

And most importantly of all; Having standards within the uplink is important. Most of the values in the uplink are just completely random. Nobody has a good grasp of what is too much or too little. Even just a hint of consistency, and people will stick to it (see implants for what I mean). And there is still some work to be done even there. A good start is weapons. Price for power can be meaningful when decided whether we want some weapons to come out more often than others. Players do enjoy making informed decisions and choices, and having affordability be a draw to some otherwise less powerful weapons (looking at you, Bulldog) can actually be a worthwhile and meaningful difference.

~~I thought it would tick off the gun nerds to change the calibers on the guns.~~

~~I also thought adding hats would be funny given the release of TF2's most recent update.~~

## Changelog
:cl:
balance: Standardizes some of the nuclear operative entries to have more consistent pricing within their respective categories.
add: Adds some new categories so that players have an easier time navigating the nuclear operative uplink.
balance: Many items have had prices reduced or adjusted to make them more desirable or more consistent within their category.
add: Weapon kits have replaced almost all the individual weapons in the uplink. You now buy these instead of the individual weapon. These often come with spare ammo or relevant gear for success.
add: Most ammo types have been standardized in price.
refactor; Removes a lot of redundant item entry code and tidies up the actual code part of the nuclear uplink so that it is much easier to find things within it.
add: Added 40 new cosmetic items to the Syndicate Store. Buy them now from the Hat Crate, only 5 TC!
code: Updated the nuclear operative uplink files.
/:cl:

### CODE REVIEW

**Comment by Helg2** [comment]
Can you also make specified mech ammo duffles? Iirc this duffle doesn't have ammo needed for mauler or something.

**Comment by carlarctg** [comment]
> _Bulldog (Moderate):_ Shotgun and three magazines of standard ammo.

Seems odd to me - Bulldog is basically the 'utility gun' and giving it base ammo only kinda goes against that. Also, I'd bump it up to Hard personally - it's kind of a pain to fight in true firefights on hallways with this close-range gun, any other weapon in the arsenal is better at that job (Even with slugs, they can't be spammed)
> _Revolver (Moderate):_ Revolver  spare magazines of standard ammo.

Similarly I feel like there's a missed opportunity here to add your cool new ammotypes.
'But walance' who cares it's nukeops. They either win instantly in blitz or die instantly in war ops.

> _L6 SAW (Moderate):_ LMG, and that's it. No spare ammo. 

Why?

> _M-90gl (Hard):_ Rifle, one spare magazine of standard ammo and a box of rubber grenades.

Rubber? Whazzat?

> _CQC (Very Hard):_ Just the book. 

Add a stealth implant and the protagonist bandana here, it'll be the big funny.

> _Double-Energy Sword (Very Hard):_ Double-energy sword, syndicate soap, antislip module, meth injector and a prisoner jumpsuit. 

Why a prisoner jumpsuit...? That's actually gonna screw you over, by the way - less armor than the turtleneck.

> _**NEW** Grenadier's Kit (Hard):_ Grenadier's belt and grenade launcher (the one that launchers chem grenades). (I replaced the shit acid grenade with another flashbang in the belt)

I've been told the grenade launcher 'sucks complete ass', so make of that what you will. What if you made it reduce grenade timers by 1 when fired?
 
> Surplus SMG (Flukie difficulty) has been unchanged. It just now comes with two rations.

Add some vodka and ushankas!

> All the various reinforcements now cost 35 TC.

Great Idea, I'm in favor.

> #### Mechs
> 
> They're just cheaper. These things still suck and they need help. They've always needed help. A slightly less excessive value for the mechs may help see people willing to spend the TC on them. I doubt it. I seriously suggest not buying these still. I keep them in primarily because they are big stompy mechs and are kind of iconic 'war ops' gear.

👍 
> Not much changed here. I standardized the implant prices to 8 TC a pop. This is in accordance with traitor implants, which ops also get. So everything in this category bar a few exceptions (like macro/microbombs) are around 8 TC. Makes sense to me, really.

Thank god, finally.

> Importantly, I made the Implant bundle 25 TC, and I unrandomized the contents. Who in the right fucking mind would spend 40 TC just to get five reviver implants is beyond me. But instead, you get one of each of the cybernetic implants except thermal eyes (you can just buy thermals and get the benefit of both vision types; x-ray and thermal vision, if you want to use smokescreens a lot).

Thank GOD.

> ## Hat Crate
> 
> **NO NEED FOR A KEY, JUST BUY IT AND PULL IT OPEN WITH YOUR STANDARD ISSUE CROWBAR!**

No keys no soul. -rep

Jebus, fix your wall of text changelog.

**Comment by DaydreamIQ** [comment]
> I've been told the grenade launcher 'sucks complete ass', so make of that what you will. What if you made it reduce grenade timers by 1 when fired?

It already does that tho? The metastation one automatically sets the timers to 3 seconds iirc

**Comment by necromanceranne** [comment]
> > _Bulldog (Moderate):_ Shotgun and three magazines of standard ammo.
> 
> Seems odd to me - Bulldog is basically the 'utility gun' and giving it base ammo only kinda goes against that. Also, I'd bump it up to Hard personally - it's kind of a pain to fight in true firefights on hallways with this close-range gun, any other weapon in the arsenal is better at that job (Even with slugs, they can't be spammed)

The free ammo is going to be consistently the basic type. While I agree that th gun is still a big weak, it is cheap. People will still going to buy it as a cheap weapon, and that is fine.

> Similarly I feel like there's a missed opportunity here to add your cool new ammotypes. 'But walance' who cares it's nukeops. They either win instantly in blitz or die instantly in war ops.

See above. I'm keeping the case ammo basic only.
 
> Why?

L6 is huge gun for one. Didn't feel right to put in a case (maybe a crate). But the other is that this weapon is already quite strong and not lacking ammo in of itself. The ammo it had is now more affordable as well. But giving is extra mags simply makes it the best gun for TC out of all of them. Overall this is both a buff (the ammo is cheaper) and a nerf (it is less tc efficient overall). Fine for such an asskicker of a gun.

> Rubber? Whazzat?

New ammo type. Bonks people like a super beanbag but it isn't good against armor. Useful for taking out unguarded limbs.

> Add a stealth implant and the protagonist bandana here, it'll be the big funny.

Tempting...
 
> Why a prisoner jumpsuit...? That's actually gonna screw you over, by the way - less armor than the turtleneck.

Tiger Co-op get depicted a lot in prisoner outfits. Felt like a Co-op kit. I don't give a fuck about jumpsuit armor, nor should anyone else. 2% means fuck all.
 
> I've been told the grenade launcher 'sucks complete ass', so make of that what you will. What if you made it reduce grenade timers by 1 when fired?

Go use it with frag grenades and you'll find out those folk were not well informed, I promise you. It _already_ minimizes the detonation timer.

> Add some vodka and ushankas!
Tempting again.

**Comment by necromanceranne** [comment]
> Can you also make specified mech ammo duffles? Iirc this duffle doesn't have ammo needed for mauler or something.

The mauler should be the second ammo duffel actually. I'll change the name so it is more obvious once I double check it has the correct ammo.

**Comment by Longestarmlonglaw** [comment]
add revolver ammo to traitor uplink?

**Comment by necromanceranne** [comment]
No.

**Comment by Longestarmlonglaw** [comment]
why, neat ammo that is always nice to have as a traitor

**Comment by necromanceranne** [comment]
I just don't have any interest in doing that.

**Comment by GuillaumePrata** [comment]
> There is a 1% chance to instagib people with direct hits from a rocket. This does the crit effect.

WTF, I never thought a day would come when I enjoy random crits...

**Comment by IndieanaJones** [comment]
Not a fan of the pre-made kits completely removing the base item choices.

**Comment by xprtdrt** [comment]
I feel like the whole weapon kit system overcomplicates something that was pretty simple to begin with, simply for the sake of change. plus i feel like i'd get confused with the kits, considering some come with ammo, some dont, some come with other shit, some come with nothing but the gun. they're categorized like they have consistency, but they all play by different rules. Maybe im just an idiot, but i find this way more of a headache to deal with over the old system of "buy just the gun then choose how much ammo, or buy the loadout kit"

The new ammo types and mech balances rule though, and im a fan of the implant kit not being random. i got fucked over by those things more than once in the past. and i will absolutely buy a hat crate next warops round.

**Comment by carlarctg** [comment]
> New ammo type. Bonks people like a super beanbag but it isn't good against armor. Useful for taking out unguarded limbs.

Oh man, you NEED to copy CM's implementation of this, it's fucking HILARIOUS. Take a look at the baton slug PR: https://github.com/cmss13-devs/cmss13/pull/1404
https://cdn.discordapp.com/attachments/745447048261795890/1037164316115341332/newHIRR.mp4

**Comment by carlarctg** [comment]
> they're categorized like they have consistency, but they all play by different rules. 

Yeah, this is something I'm iffy about as well. I agree with trying to standarize the guns and nukies overall more as it literally takes indecisive people (me) the full 20 warops minutes to decide what the frick to get. However, I think this style is pretty nauseating as well, and I'd rather they were balanced around what a 'full kit' is rather than the 3 static TC tiers, if that makes sense?

**Comment by githubuser4141** [comment]
Would it be possible to either add armor packages for the mechs, or - buff the mechs' innate armour and remove their armour slots? Mauler/Marauder mechs arrive with very barebones anti-melee armour (50%), and a whopping 4 (!!!) free armor slots. However, the only way for Nukies to upgrade their armour is by going to robotics, doing the mech experiment and printing armor modules which in the few times I've seen a nukie mech I've never ever seen it happen? Nukies seem to have the mindset that the only way of interacting with the station is by shooting or eswording it; when's the last time you saw a nukie going "Oh I'd better do the human dissection so I can research mechs & print more armour modules for my Mauler."?

**Comment by Loafin34** [comment]
you know what'd be cool: some sprites for these hats you yap about. also meth injector WILL kill you from brain damage please add some mannitol

**Comment by HoodMilk** [comment]
The change to grenades is kind of lame, instead of using some leftover TC to buy a grenade or two you're forced to purchase them in 15 tc bundles. I don't even see the point of putting them into bundles, clicking the buy button three times is not difficult or confusing to a new player.

**Comment by SpessManCN** [comment]
is the rubber nade like that CM13 rubber nade got stun?

**Comment by Fikou** [comment]
> you know what'd be cool: some sprites for these hats you yap about. also meth injector WILL kill you from brain damage please add some mannitol

it does have mannitol and the hats are already in the game

**Comment by necromanceranne** [comment]
> Not a fan of the pre-made kits completely removing the base item choices.

I feel you'll have to elaborate as to why, because I'm not sure why you wouldn't want ammo with your gun. I don't know a single player who only buys the gun and not additional ammo. So why give them the choice to underprepare themselves? For many of these weapons, the cost effectiveness has gone up, and some, mostly the same.

> I feel like the whole weapon kit system overcomplicates something that was pretty simple to begin with, simply for the sake of change. 

I think you will need to elaborate more here as to how it overcomplicates things. If you can open toolboxes, you can utilize the gun cases given to you. It's as simple as that. Many uplink items come in containers like this. If you have been using mosins in cargo recently, you'll find that you have already been interacting with these containers, as the code is repurposed from there. 

My explanation is very simply as well. 'I think it would be good to give players the necessary bits and pieces they were otherwise going to be getting, but mostly _for free/frontloaded in the cost_"

**Comment by necromanceranne** [comment]
> Would it be possible to either add armor packages for the mechs, or - buff the mechs' innate armour and remove their armour slots? Mauler/Marauder mechs arrive with very barebones anti-melee armour (50%), and a whopping 4 (!!!) free armor slots. However, the only way for Nukies to upgrade their armour is by going to robotics, doing the mech experiment and printing armor modules which in the few times I've seen a nukie mech I've never ever seen it happen?
> 
> Nukies seem to have the mindset that the only way of interacting with the station is by shooting or eswording it; when's the last time you saw a nukie going "Oh I'd better do the human dissection so I can research mechs & print more armour modules for my Mauler."? It's also kinda unintuitive for newbies that the nukie mechs require upgrades from stationside exosuit fabs in order to be halfway decent.

I don't think we expect nukies to be doing that, and I certainly wouldn't have thought whoever changed mech gear slots to have been thinking of nukie mechs at all (everyone forgets about them really, they get overlooked a lot).

I don't want to get too into mech design and balance in this PR but I'll maybe look at it later. I think there are more fundamental issues with mechs as a whole than just equipment.

**Comment by SimplySmiley123** [comment]
> > Not a fan of the pre-made kits completely removing the base item choices.
> 
> I feel you'll have to elaborate as to why, because I'm not sure why you wouldn't want ammo with your gun. I don't know a single player who only buys the gun and not additional ammo. So why give them the choice to underprepare themselves? For many of these weapons, the cost effectiveness has gone up, and some, mostly the same.


Quite a few people buy the gun as a sidearm (similiar to marines on TGMC taking their loadout then grabbing a random wahtever gun to run out with and use till they run out of ammo and whip out their primary weapon), mostly the pistols or the SMG with leftover TC, while focusing on another weapon.
Same with ammo and explosives. You buy more with spare TC instead of getting a whole bundle.

Also to be honest, the P-90 wannabe is something I never seen used due to being pretty much same as the SMG in terms of what it does and the damage of it but with raised price tag and grenades which you rarely ever get to use and have them do something and also add to the overall price of the weapon.
Most nukies I seen who would want that glorified grenade launcher would just buy the cr20 and explosives (grenades, bombs, whatever) separately because it's just not worth it to get the M-90gl.
Or go full in on explosive and buy the grenadier belt and spend rest of TC on more explosives.
And yeah, the grenadel auncher does suck royally even with actual lethal grenades, it's more of a hassle for something you can just do by throwing just as well.

**Comment by atlasle** [comment]
can you summarize at all this pr is just one giant fucking wall of text

**Comment by necromanceranne** [comment]
> Quite a few people buy the gun as a sidearm (similiar to marines on TGMC taking their loadout then grabbing a random wahtever gun to run out with and use till they run out of ammo and whip out their primary weapon), mostly the pistols or the SMG with leftover TC, while focusing on another weapon. Same with ammo and explosives. You buy more with spare TC instead of getting a whole bundle.

I'm not sure what gun you're referring too. If you mean to suggest that the reason you don't want the cases to give you _free_ ammo is because it is meant to be a sidearm, then I'd ask how exactly the free ammo that now comes with the majority of these guns matters much at all? And additionally, why are you spending the TC on a full new gun beyond your main weapon when you are given a workable sidearm with the Ansem, which is extra free and the ammo is in abundance on the base?

I feel making the decision to spend as much as 13-15 TC (currently) on two bulky weapons, but only using one of them _sometimes_ is...not very good decision-making and I'm sorry the system has convinced you otherwise. If that is still your intent after this change, how is that not made stronger by giving you more ammo when you make those purchases?
 
> Also to be honest, the P-90 wannabe is something I never seen used due to being pretty much same as the SMG in terms of what it does and the damage of it but with raised price tag and grenades which you rarely ever get to use and have them do something and also add to the overall price of the weapon. 

I mean I already talked to fikou about the M-90gl and I made the argument to him that it kind of has no place at the moment, so I agree that it probably is able to be cut and the grenade launcher moved somewhere else, I think Viro still wants to make sprites for it. So I'd much rather just accept it being there and work out a good niche for it than get rid of it at the moment.

That said, I actually disagree about the grenades entirely, and I think you undervalue direct impact explosions immensely. The grenade ammo being much cheaper should encourage people to get it more often, I would hope. I'm not even really sure if they'll keep the rubber grenades when I could just make it standard grenades. Maybe they could get both.

**Comment by necromanceranne** [comment]
> can you summarize at all this pr is just one giant fucking wall of text

The guns come with ammo now. Ammo is cheaper. Weapons and gear have different prices. Hats.

**Comment by SomeoneYouProbablyKnow** [comment]
This doesn't really reduce customization, does it? The bundles are less expensive than the actual items are right now

Like, i already like the grenade belt and now it's cheaper and comes with a launcher i'm not gonna use because i like yeeting nades.

If you wanna use like a bulldog for funny door rounds you still can. Now it comes with a few mags just cos

It's like complaining "thrown weapons" have both bolas and ninja stats, what if you wanted to use just the stars.

If anything bundles might inspire and nudge players to experiment. Like the idea of adding the stealth implant to cqc gonna make some people use the box who wouldn't have otherwise.
gimmick idea

**Comment by Serxule** [comment]
> This doesn't really reduce customization, does it? The bundles are less expensive than the actual items are right now

> If anything bundles might inspire and nudge players to experiment. Like the idea of adding the stealth implant to cqc gonna make some people use the box who wouldn't have otherwise. gimmick idea

you're under the assumption that anyone whose buying x is only going to use x, ignoring misc gear, and will just buy x and ammo for it

**Comment by necromanceranne** [comment]
> Looks great, though I'm not sure if 100% standardising items shared with traitors pricewise makes the most sense? Traitor items are rarely balanced pricewise around going mano-a-mano with 60 crew in open warfare

Nothing here touches traitor items. These are entirely unique to nukies. Including the revolver and desword.

**Comment by necromanceranne** [comment]
> you're under the assumption that anyone whose buying x is only going to use x, ignoring misc gear, and will just buy x and ammo for it

Well if they buy any individual item at all, they get some ammo/gear now that synergies with it, so what's the problem? It doesn't lock you out of additional purchases or buying other guns.

**Comment by Serxule** [comment]
> > you're under the assumption that anyone whose buying x is only going to use x, ignoring misc gear, and will just buy x and ammo for it
> 
> Well if they buy any individual item at all, they get some ammo/gear now that synergies with it, so what's the problem? It doesn't lock you out of additional purchases or buying other guns.
> 
> I feel like the person making assumptions might be yourself, and I'd invite you to have another read over the PR.

it still costs more if you just want the gun and no ammo unless you can still buy stuff without the bundles, what if i want to buy a bunch of .357s? thats 2tc more per for ammo i dont even want because i just want to do a gimmick

**Comment by necromanceranne** [comment]
You have an entire station of people of people to shoot and two guns to reload. Are you just hoping to kill everyone in 14 shots?

**Comment by Serxule** [comment]
> You have an entire station of people of people to shoot and two guns to reload. Are you just hoping to kill everyone in 14 shots? The old speedloaders were 4 TC a pop.

mainly referring to warops when you can afford like 10

**Comment by necromanceranne** [comment]
I now given 100% less of a shit about your overly specific and niche consideration.

**Comment by Loafin34** [comment]
> > > you're under the assumption that anyone whose buying x is only going to use x, ignoring misc gear, and will just buy x and ammo for it
> > 
> > 
> > Well if they buy any individual item at all, they get some ammo/gear now that synergies with it, so what's the problem? It doesn't lock you out of additional purchases or buying other guns.
> > I feel like the person making assumptions might be yourself, and I'd invite you to have another read over the PR.
> 
> it still costs more if you just want the gun and no ammo unless you can still buy stuff without the bundles, what if i want to buy a bunch of .357s? thats 2tc more per for ammo i dont even want because i just want to do a gimmick

tough luck man? i wanted to get lucky with my implant box and get 8 microbombs but it's not in the cards.

**Comment by necromanceranne** [comment]
I've made the CQC manual come with a stealth implant and bandana. 

The sniper rifle has two normal mags instead of one, and only one disruptor.

M-90gl comes with two magazines of normal ammo and not one.

~~Maybe there is a chance at an unusual hat when you open the crate~~

**Comment by necromanceranne** [comment]
> I do not like this kit stuff. I like mixing and matching. That's the purpose of the uplink.

This change should not impact this. The kits are, mostly, free gear on top of existing gun purchases. They're not a full kit and not take all your TC.

You should, ideally, have more TC to use than before.

**Comment by nukularpower** [comment]
since this touches the lab keys, i will add:  pleeeeeease add upgraded parts to the nukie xenobio machines, like at least tier 2 or 3

**Comment by CCC23** [comment]
CROCKET!

**Comment by weesnaw-ops** [comment]
> > I do not like this kit stuff. I like mixing and matching. That's the purpose of the uplink.
> 
> This change should not impact this. The kits are, mostly, free gear on top of existing gun purchases. They're not a full kit and not take all your TC. It is, for almost all the kits, the gun and some free ammo. Or things you, as an op using this, are strongly encouraged to take. The ammo itself is cheaper if bought separately.
> 
> You should, ideally, have more TC to use than before. Assuming you're not buying 10 revolvers.

If this is true then sure this change is fine

**Comment by CRITAWAKETS** [comment]
Same price for purchasing two weapons actually makes this a pretty damn good buff, even if you're dual-wielding a weapon (since it'll cost the same, but you'll get more ammo and kit things with it)

I'm in favor of this, kits just allow way more variety and interesting stuff since they can be stuffed with non-uplink items, alongside helping new players know that they should bring ammo.

**Comment by Sparkezel** [comment]
Why the hell there are rubber grenades in m90gl? 
Whats the point of bundling everything if you don't show what ammo the L6 uses?
Do the surplus smgs come in bundles or do they just spawn in your hand? They costed 1TC right? It was funny to buy shoot and drop them.
If you keep L6 the way you do rn, I think you should be able to buy all guns alone as well, slightly cheaper than the bundle.

**Comment by necromanceranne** [comment]
> Why the hell there are rubber grenades in m90gl? 

I'm going to develop on the idea a bit more, but mostly for bonking people in the head at high velocity.

>Whats the point of bundling everything if you don't show what ammo the L6 uses? 

I could probably make it more obvious on the gun and uplink description, but..

>If you keep L6 the way you do rn, I think you should be able to buy all guns alone as well, slightly cheaper than the bundle.

I don't want to drop prices down even further than what it is currently (most items didn't move much in price point), and additionally, the point is to ensure players have the tools necessary for success. This is meant for newer players and players new to nukies. I'm not interested in separating the weapon from the cases. That ultimately undermines the intent of this change.

>Do the surplus smgs come in bundles or do they just spawn in your hand? They costed 1TC right? It was funny to buy shoot and drop them. 

The gun is 2 TC currently because I believe people noted that the ammo was the same price as the gun. When I moved it to a kit I didn't add ammo so as to maintain this. It instead comes with rations.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/crates_lockers/crates/large.dm [hunk 1]
```diff
@@ -55,3 +55,18 @@
...
+		new /obj/effect/spawner/random/clothing/funny_hats(src)
+	for (var/i in 1 to 5)
+		new /obj/item/mod/module/hat_stabilizer(src)
+	if(prob(1))
+		var/our_contents = list()
+		for(var/obj/item/clothing/head/any_hat in contents)
+			our_contents[any_hat]++
+		if(our_contents)
+			var/obj/item/clothing/head/lucky_hat = pick(our_contents)
+			lucky_hat.AddComponent(/datum/component/unusual_effect, color = "#FFEA0030", include_particles = TRUE)
```
Comment (Jacquerel) [inline code/game/objects/structures/crates_lockers/crates/large.dm:71]: I had no idea we already had this component

File: code/game/objects/items/storage/toolbox.dm [hunk 2]
```diff
@@ -313,24 +313,136 @@
...
 	icon = 'icons/obj/storage/case.dmi'
-	icon_state = "mosin_case"
+	icon_state = "infiltrator_case"
 	lefthand_file = 'icons/mob/inhands/equipment/toolbox_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/equipment/toolbox_righthand.dmi'
-	inhand_icon_state = "mosin_case"
+	inhand_icon_state = "infiltrator_case"
 	has_latches = FALSE
+	var/gun_to_spawn = /obj/item/gun/ballistic/automatic/pistol
+	var/ammo_to_spawn = /obj/item/ammo_box/magazine/m9mm
```
[the remark this answers is not in the document]
Comment (necromanceranne) [reply code/game/objects/items/storage/toolbox.dm:326]: Fair. I kind of named the vars in another pr where it made sense and didn't fix it here.

File: code/game/objects/structures/crates_lockers/crates/large.dm (hunk 1 above)
Comment (necromanceranne) [reply code/game/objects/structures/crates_lockers/crates/large.dm:71]: Melbert showed me it, haha. It needs more visual affects however. Right now it is very barebones.
